lib/sprockets/loader.rb

```diff
@@ -131,16 +131,22 @@ def load_from_unloaded(unloaded)
 
         # Read into memory and process if theres a processor pipeline
         if processors.any?
-          result = call_processors(processors, {
-            environment: self,
-            cache: self.cache,
-            uri: unloaded.uri,
-            filename: unloaded.filename,
-            load_path: load_path,
-            name: name,
-            content_type: type,
-            metadata: { dependencies: dependencies }
-          })
+          processor_interface = Hash.new do |hash, key|
+            if :source_path == key
```

Comment:
So...I used "filename" in this guide https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#supporting-all-versions-of-sprockets-in-processors is it wrong?

In 4.x we have

```ruby
{
            environment: self,
            cache: self.cache,
            uri: unloaded.uri,
            filename: unloaded.filename,
            load_path: load_path,
            source_path: source_path,
            name: name,
            content_type: type,
            metadata: {
              dependencies: dependencies,
              map: []
            }
          }
```

In 3.x we have

```ruby
{
            environment: self,
            cache: self.cache,
            uri: unloaded.uri,
            filename: unloaded.filename,
            load_path: load_path,
            name: name,
            content_type: type,
            metadata: { dependencies: dependencies }
          }
```

It doesn't look like "filepath" was a thing? Also looks like I need to add `source_path` to 3.x. So that. Now I need to figure what exactly `source_path` is in 4.x.

I think a key was changed and removed in a previous version of 3.x by mistake? My compare-view fu is failing me right now.